Issue/8 host header

iperdomo opened the pull request.

iperdomo committed:
[#8] Honor `Host` header present in request

The decision is to fallback to the `Host` http header if is present in
the request to calculate the proper location where to `PATCH` chunks

We also take into account of there are `X-Forwarded` headers to build
the resulting location. If `X-Forwarded-Proto` is present it takes
precedence otherwise we fallback to the `:scheme` in the request

We attempt to remove default ports for different protocols: 80 for http
and 443 for https

iperdomo committed:
[#8] Update build script

iperdomo committed:
[#8] Update Travis-CI config

iperdomo committed:
[#8] Test with openjdk8

iperdomo committed:
Merge branch 'master' into issue/8-host-header

iperdomo committed:
[#8] Remove configuration based on user profile

- We use the plugin directly under the :dev profile

iperdomo committed:
[#8] Improve logic on build|release scripts

dlebrero commented on test/org/akvo/resumed_test.clj:
```diff
@@ -211,18 +211,30 @@
-               :uri     "/path"}]
-      (is (= "https://www.akvo.org/path"
-             (get-location req))))))
+               :uri     "/path"
```
inline the req, req-2, req-3, ... to improve readability.

dlebrero commented on src/org/akvo/resumed.clj:
```diff
@@ -120,26 +120,52 @@
-        f-host (or (get-header req "x-forwarded-host")
-                   (get-header req "host"))
-        f-proto (get-header req "x-forwarded-proto")
+  (let [host (get-host req)
```
inline?, the fn names are good, I do not see the need for a `let` anymore. Maybe also rename "get-host" to "host", "get-port" to "port", ...

dlebrero commented on src/org/akvo/resumed.clj:
```diff
@@ -120,26 +120,52 @@
+  [req]
+  (let [forwarded-proto (get-header req "x-forwarded-proto")
+        known-protocols #{"http" "https"}]
+    (if (or (str/blank? forwarded-proto)
```
str/blank? is redundant, as it is included in the (not (known-protocols ...

dlebrero commented on src/org/akvo/resumed.clj:
```diff
@@ -120,26 +120,52 @@
+                 (get-header req "origin")
+                 (first (.split (or (get-header req "host") "") ":")))]
+    (if (str/blank? host)
+      (:server-name req)
```
Maybe we can move this to the last thing in the "or" in line 127, if like 129 returns a nil, which it could do if we used a regular expression to extract the host, or `some->`

dlebrero commented on src/org/akvo/resumed.clj:
```diff
@@ -120,26 +120,52 @@
+  [req]
+  (let [forwarded (or (get-header req "x-forwarded-host")
+                      (get-header req "x-forwarded-proto"))
+        host (get-header req "host")]
```
move this `host` to an if-let in line 157, so the context of the host is smaller (I do not need to parse it in the first part of the if condition), plus we remove the and in line 157.

iperdomo committed:
[#8] Inline request map to improve readability

kardan reviewed (approved):
Besides already mentioned code changes I agree with this batch of changes.

iperdomo committed:
[#8] Apply code-review changes

iperdomo commented:
@dlebrero changes made

iperdomo committed:
[#8] Code cleanup - Positive conditional

Extract the set of known protocols to a constant to avoid creating
a new one each time.

Use (if ,,,) instead (if (not ,,,)) to
